src/go/rdctl/pkg/utils/utils.go

```diff
@@ -1,91 +1,49 @@
- * And mode bits don't make sense on Windows.
- */
-func CheckExistence(candidatePath string, modeBits fs.FileMode) string {
-	stat, err := os.Stat(candidatePath)
-	if err != nil {
-		return ""
-	}
-	if modeBits != 0 && (!stat.Mode().IsRegular() || stat.Mode().Perm()&modeBits == 0) {
-		// The modeBits check is only for executability -- we only care if at least one of the three
-		// `x` mode bits is on. So this check isn't used for a general permission-mode-bit check.
-		return ""
-	}
-	return candidatePath
-}
-
-// Returns the absolute path to the Rancher Desktop executable.
-// Returns an empty string if the executable was not found.
-func GetWindowsRDPath(rdctlPath string) string {
-	if rdctlPath != "" {
-		normalParentPath := MoveToParent(rdctlPath, 5)
-		candidatePath := CheckExistence(filepath.Join(normalParentPath, "Rancher Desktop.exe"), 0)
-		if candidatePath != "" {
-			return candidatePath
-		}
+// Verify that the candidatePath is usable as a Rancher Desktop "executable". This means:
+//   - check that candidatePath exists
+//   - if checkExecutability is true, check that candidatePath is a regular file,
+//     and that it is executable
+//
+// Note that candidatePath may not always be a file; in macOS, it may be a
+// .app directory.
+func checkUsability(candidatePath string, checkExecutability bool) (bool, error) {
+	statResult, err := os.Stat(candidatePath)
+	if errors.Is(err, fs.ErrNotExist) {
+		return false, nil
 	}
-	homedir, err := os.UserHomeDir()
 	if err != nil {
-		homedir = ""
+		return false, fmt.Errorf("failed to get info on %q: %w", candidatePath, err)
 	}
-	dataPaths := []string{}
-	// %LOCALAPPDATA%
-	dir, err := directories.GetLocalAppDataDirectory()
-	if err == nil {
-		dataPaths = append(dataPaths, dir)
-	}
-	// %APPDATA%
-	dir, err = directories.GetRoamingAppDataDirectory()
-	if err == nil {
-		dataPaths = append(dataPaths, dir)
+
+	if !checkExecutability {
+		return true, nil
 	}
-	// Add these two paths if the above two fail to find where the program was installed
-	dataPaths = append(
-		dataPaths,
-		filepath.Join(homedir, "AppData", "Local"),
-		filepath.Join(homedir, "AppData", "Roaming"),
-	)
-	for _, dataDir := range dataPaths {
-		candidatePath := CheckExistence(filepath.Join(dataDir, "Programs", "Rancher Desktop", "Rancher Desktop.exe"), 0)
-		if candidatePath != "" {
-			return candidatePath
-		}
+
+	if !statResult.Mode().IsRegular() {
+		return false, nil
 	}
-	return ""
-}
 
-func GetDarwinRDPath(rdctlPath string) string {
-	if rdctlPath != "" {
-		// we're at .../Applications/R D.app (could have a different name)/Contents/Resources/resources/darwin/bin
-		// and want to move to the "R D.app" part
-		RDAppParentPath := MoveToParent(rdctlPath, 6)
-		if CheckExistence(filepath.Join(RDAppParentPath, "Contents", "MacOS", "Rancher Desktop"), 0o111) != "" {
-			return RDAppParentPath
-		}
+	if statResult.Mode().Perm()&0o111 == 0 {
```

Comment:
It looks like on Unix platforms doing [`Access(X_OK)`](https://pkg.go.dev/golang.org/x/sys/unix#Access) might be more proper, but the problem with that is it wouldn't compile on Windows… so we'd need to split to `utils_windows.go` and `utils_unix.go`. (This might be an issue, for example, if the executable is owned by a different user with `a+r u+x`.)